eng/source-build-patches/0002-Remove-apphost-prebuilt-usage.patch

```diff
@@ -0,0 +1,73 @@
+From e2f6d8d0ff79a23433d205b9eb0c7526949bbe6b Mon Sep 17 00:00:00 2001
```

Comment:
This analyzer ships with SDK don't we want to build it as well?

Reply:
Hm... it does? Can you point me to an SDK that includes it? I am looking at 5.0.103 and I can't find the analzyer:

```
$ tar tf dotnet-microsoft-built-sdk-5.0.103.tar.gz | grep -i illink
./sdk/5.0.103/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.ILLink.targets
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/Sdk/
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/Sdk/Sdk.props
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/Icon.png
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/net472/
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/net472/System.Collections.Immutable.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/net472/ILLink.Tasks.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/net472/System.Reflection.Metadata.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/net472/Mono.Cecil.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/illink.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/illink.deps.json
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/ILLink.Tasks.deps.json
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/illink.runtimeconfig.json
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/ILLink.Tasks.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/Mono.Cecil.dll
./sdk/5.0.103/Sdks/Microsoft.NET.ILLink.Tasks/tools/netcoreapp3.0/Mono.Cecil.Pdb.dll
```

Reply:
It's a new in .NET6

Reply:
Thanks! I am not sure how to handle this. Any ideas @crummel, @dagood @dseefeld?

Reply:
I think we should merge this PR as-is, and fix up the patches and this functional issue as followup. This PR represents the state of dotnet/source-build, it's not necessary to solve every source-build issue right here and now.

FYI @dseefeld this will also need to be fixed up in the old-infra-style .NET 6 preview1 work, once we find a resolution.